src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -1367,7 +1370,7 @@ public ErrorDiagnostic UnknownModuleReferenceScheme(string badScheme, ImmutableA
             public ErrorDiagnostic ParamOrOutputResourceTypeUnsupported() => new(
                 TextSpan,
                 "BCP231",
-                "Using resource-typed parameters and outputs requires enabling EXPERIMENTAL feature BICEP_RESOURCE_TYPED_PARAMS_AND_OUTPUTS_EXPERIMENTAL.");
+                "Using resource-typed parameters and outputs requires enabling EXPERIMENTAL feature resourceTypedParamsAndOutputs.");
```

Comment:
[nit] Add quotes around `resourceTypeParamsAndOutputs`. Is there a const defined anywhere that we could use to refer to the feature name?